client/sftp_client.py

```diff
@@ -56,100 +93,429 @@ def signal_sftp_start(self, _):
-		col.add_attribute(col_text, "text", 0)
+		col_img = Gtk.CellRendererPixbuf()
+		col = Gtk.TreeViewColumn('Direction')
+		col.pack_start(col_img, False)
+		col.add_attribute(col_img, 'pixbuf', 0)
 		self.treeview_transfer.append_column(col)
-		self._tvmodel = Gtk.ListStore(str)
-		self.treeview_transfer.connect('size-allocate', self._treeview_changed)
-		self.treeview_transfer.set_model(self._tvmodel)
-		self.treeview_transfer.connect('button_press_event', self._signal_treeview_button_pressed)
+
+		self.treeview_transfer.append_column(get_treeview_column('Local File', col_text, 1, m_col_sort=1, resizable=True))
+		self.treeview_transfer.append_column(get_treeview_column('Remote File', col_text, 2, m_col_sort=2, resizable=True))
+		self.treeview_transfer.append_column(get_treeview_column('Status', col_text, 3, m_col_sort=3, resizable=True))
+
+		col_bar = Gtk.TreeViewColumn('Progress')
+		progress = Gtk.CellRendererProgress()
+		col_bar.pack_start(progress, True)
+		col_bar.add_attribute(progress, 'value', 4)
+		col_bar.set_property('resizable', True)
+		self.treeview_transfer.append_column(col_bar)
+
+		self.treeview_transfer.append_column(get_treeview_column('Size', col_text, 5, m_col_sort=3, resizable=True))
+		self._tv_model = Gtk.TreeStore(GdkPixbuf.Pixbuf, str, str, str, int, str)
+		self.treeview_transfer.connect('size-allocate', self.signal_tv_size_allocate)
+		self.treeview_transfer.connect('button_press_event', self.signal_tv_button_pressed)
+		self.treeview_transfer.connect('row-expanded', self.signal_tv_expand_row)
+
+		self.treeview_transfer.set_model(self._tv_model)
+		self.treeview_transfer.show_all()
 
 		self.popup_menu = Gtk.Menu.new()
+
+		self.menu_item_paused = Gtk.CheckMenuItem.new_with_label('Paused')
+		menu_item = self.menu_item_paused
+		menu_item.connect('toggled', self.signal_menu_toggled_paused)
+		self.popup_menu.append(menu_item)
+
+		self.menu_item_cancel = Gtk.MenuItem.new_with_label('Cancel')
+		menu_item = self.menu_item_cancel
+		menu_item.connect('activate', self.signal_menu_activate_cancel)
+		self.popup_menu.append(menu_item)
+
+		menu_item = Gtk.SeparatorMenuItem()
+		self.popup_menu.append(menu_item)
+
 		menu_item = Gtk.MenuItem.new_with_label('Clear')
-		menu_item.connect('activate', self.log_clear)
+		menu_item.connect('activate', self.signal_menu_activate_clear)
 		self.popup_menu.append(menu_item)
 		self.popup_menu.show_all()
 
-	def _signal_treeview_button_pressed(self, _, event):
+	def _get_selected_tasks(self):
+		treepaths = self._get_selected_treepaths()
+		if treepaths is None:
+			return None
+		# Find the task selected task then find tall the tasks in the queue with it as a parent.
+		parent_task = [task for task in self.queue.queue if task.treerowref is not None and task.treerowref.valid() and task.treerowref.get_path() in treepaths][0]
+		children = [task for task in self.queue.queue if task.parents is not None and parent_task in task.parents]
+		parent_task = [parent_task,]
```

Comment:
This should just be simply `children.insert(0, parent_task)` so you don't have to create a new list.